src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactoryCompilers.cs

```diff
@@ -14,7 +15,7 @@ namespace Microsoft.Build.Tasks
     internal abstract class RoslynCodeTaskFactoryCompilerBase : ToolTaskExtension
     {
 #if RUNTIME_TYPE_NETCORE
-        private static readonly string DotnetCliPath = System.Diagnostics.Process.GetCurrentProcess().MainModule.FileName;
+        private readonly string dotnetCliPath;
```

Comment:
our convention is that `private` fields should start with an `_`

```suggestion
        private readonly string _dotnetCliPath;
```

(we don't do this universally everywhere but it's used in this file, for example `_executablePath`, so let's stick with it here)